docs/scripting/trust_scripts/sys.upgrades.mdx

```diff
@@ -0,0 +1,111 @@
+---
+title: sys.upgrades
+---
+
+> In-game description of the script or function
+
+### Security Level
+
+HIGHSEC
+
+## Syntax
+
+### CLI
+
+```
+sys.upgrades
+```
+
+### Script
+
+```
+#hs.scripts.upgrades()
+```
+
+### Parameters
+
+#### filter
+
+The 'filter' parameter takes an object which can specify the properties 'loaded' and 'chars'.
+
+- loaded
+
+Takes a boolean, specifies whether to display loaded upgrades with true, or unloaded upgrades with false.
+
+- chars
+
+Takes MongoDB query compatibles, e.g. `chars: { "$gte" : 2000 }`
```

Comment:
This is an interesting thing but I think you've covered it already. It takes more than loaded and chars. It should take conceivably any key that is in an upgrade. It let's use the mongo operators on your search but there are some fields that do not work for whatever reason that I cannot recall.

So the interesting bit is laying out ...sub...arguments? in the template or style guide as far as handling it or if we even want to go into that detail here instead of a page like [[mongo operators]]

Reply:
For sure, this one was a bit tricky. I think eventually I'd prefer the [[mongo operators]] suggestion but I'm not sure if we want to do that in phase 1 or in a later stage of production. Not sure how to proceed here.